api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -17,6 +18,23 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Wasm]
 // [#extension: envoy.bootstrap.wasm]
 
+// If there is a fatal error on the VM (e.g. exception, abort()), then the policy will be applied.
+enum FailurePolicy {
+  // No policy is specified. The default policy will be used. The default policy is ``FAIL_RELOAD``.
+  UNSPECIFIED = 0;
+
+  // All plugins associated with the VM will be reloaded then new requests are processed. This
+  // make sense when the VM failure is caused by runtime exception, abort(), or panic().
```

Comment:
What if the failure is caused by something else?  If we think this behavior makes sense only in these cases, can we trigger the behavior only in these cases, rather than doing it for any failure?  Or is there no down-side to doing this on other types of failures?

Reply:
I think we can do this for runtime error first. And to add new failure support in the future if necessary.